Account.java

```diff
@@ -0,0 +1,86 @@
+package accounting;
+
+import agents.Agent;
+import actions.Action;
+import contracts.Contract;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class Account {
+
+    private Account(String name, AccountType accountType, Double startingBalance) {
+        this.name = name;
+        this.accountType = accountType;
+        this.balance = startingBalance;
+        this.contractClass = null;
+        this.contracts = new HashSet<>();
+    }
+
+    Account(String name, AccountType accountType) {
+        this(name,accountType,0.0);
+    }
+
+    private double balance;
+
+//    private Collateral collateralType;
+    private AccountType accountType;
+    private String name;
+    private Class<? extends Contract> contractClass;
+    protected HashSet<Contract> contracts;
+
+
+    void addContract(Contract contract) {
```

Comment:
is it intentional that these functions scope are the Java default? It's likely that's best, but I didn't know if some, like addContract, should be public?